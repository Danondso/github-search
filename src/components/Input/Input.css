.input {
    height: 2rem;
    border:none;
    border-radius: 8px;
    font-size: large;
    color: #000;
    background-color: #fff;
    text-align: center;
}

::placeholder {
    color: #000;
    text-align: center;
}

input:focus{
    outline: none;
}