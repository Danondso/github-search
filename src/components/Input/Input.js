import React from 'react';
import PropTypes from 'prop-types';
import './Input.css';

function Input({
  placeholder,
  onChange,
  disabled,
}) {
  return (
    <input
      className="input"
      disabled={disabled}
      placeholder={placeholder}
      onChange={onChange}
    />
  );
}

Input.propTypes = {
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
};

Input.defaultProps = {
  disabled: false,
  onChange: undefined,
};

export default Input;
