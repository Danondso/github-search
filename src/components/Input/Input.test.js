import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from './Input';

const onChangeMock = jest.fn();
beforeEach(() => {
  onChangeMock.mockReset();
});

const placeholderText = 'Test Input';

describe('Input Test Suite', () => {
  test('Input is clickable', () => {
    render(
      <Input
        onChange={onChangeMock}
        placeholder={placeholderText}
      />,
    );
    const inputElement = screen.getByPlaceholderText(placeholderText);
    expect(inputElement).toBeInTheDocument();
  });

  test('onChangeMock is called when text is input', () => {
    render(
      <Input
        onChange={onChangeMock}
        placeholder={placeholderText}
      />,
    );
    const text = 'test';
    userEvent.type(screen.getByPlaceholderText(placeholderText), text);
    expect(onChangeMock.mock.calls[0][0].target.value).toEqual(text);
    expect(onChangeMock).toHaveBeenCalledTimes(4);
  });

  test('Text cannot be input when disabled', () => {
    render(
      <Input
        onChange={onChangeMock}
        placeholder={placeholderText}
        disabled
      />,
    );
    userEvent.type(screen.getByPlaceholderText(placeholderText), 'test');
    expect(onChangeMock).toHaveBeenCalledTimes(0);
  });
});
