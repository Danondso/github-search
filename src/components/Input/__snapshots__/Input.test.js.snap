// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Test Suite Input is clickable 1`] = `
<input
  class="input"
  placeholder="Test Input"
/>
`;
